backend/algorithms/objects/conditions.py

```diff
@@ -265,7 +309,11 @@ def __init__(self, exclusion: str):
         self.exclusion = exclusion
 
     def validate(self, user: User) -> tuple[bool, list[str]]:
-        return not user.has_taken_course(self.exclusion), []
+        is_valid = not user.has_taken_course(self.exclusion)
+        return is_valid, [] if is_valid else [f"{self.exclusion} is an exclusion course for this one"]
```

Comment:
Function typing of tuple[bool, list[str]] isn't always true since it's sometimes just list[str] - is it possible to always return the same type or not really?